static/js/components/event-page.js

```diff
@@ -1,22 +1,16 @@
 import React from 'react';
 import client from '../api';
+import {Link, useParams} from "react-router-dom";
 
-export default class EventPage extends React.Component {
-    constructor(props) {
-        super(props);
 
-        this.exportEvent = this.exportEvent.bind(this);
-    }
-    exportEvent(e) {
+function exportEvent(eventID) {
         // Retrieves event information and returns as ical file
-        var eventID = this.props.event.id;
         var email = document.getElementById("exportEmail").value;
 
         const emailPattern = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/
 
         if (!emailPattern.test(email)) {
           alert("You must enter an email!");
-          e.preventDefault();
```

Comment:
Why did you remove this? With it gone, if you submit an incorrect email, the form will still submit. `e.preventDefault` prevents the form from submitting, which is the default behavior of the button.